docs-v2/design_proposals/ko-builder-auto-sync.md

```diff
@@ -0,0 +1,170 @@
+
+3.  When a change event takes place for local source code files, determine
+    the platform for the rebuilt binary. Use the field from `skaffold.yaml` or
+    `--platform` flag if present, if not, default to `linux/<host arch>`. Using
+    the host architecture as the default helps Minikube and KinD users.
+
+4.  Set the [`KOCACHE`](https://github.com/ko-build/ko/issues/264)
+    environment variable if unset, so that ko builds the binary in a
+    deterministic location. A clear contract from ko on this behavior would be
+    helpful.
+
+5.  When constructing the
+    [ko build options](https://github.com/ko-build/ko/blob/main/pkg/build/options.go),
+    ensure that ko doesn't download the base image again. Skaffold is only
+    rebuilding the binary, so the base image isn't required. To achieve this,
+    provide an
+    [empty image](https://github.com/google/go-containerregistry/tree/main/pkg/v1/empty)
+    as the ko base image.
+
+6.  Use ko to build the new binary
+    [src](https://github.com/ko-build/ko/blob/5e0452ad67230076340d0e28dd8488e4370675c2/pkg/build/gobuild.go#L967).
+
+7.  Use the existing Skaffold sync feature to sync the rebuilt binary from
+    the local file `$KOCACHE/bin/<import path>/linux/<arch>/out` to the
+    container destination `/ko-app/<base name of import path>`.
+
+## Open questions
+
+1.  Should `sync.auto` default to `true` for the ko builder, as it does for
+    the Buildpacks and Jib builders?
+
+    Users must specify a default base image that works with watchexec, and
+    the default ko base image doesn't. This means that if we default to
+    `true`, `skaffold dev` will fail for ko users who don't specify a
+    compatible base image.
+
+    Options:
+
+    -   Default to `true` for consistency with other builders. If
+        skaffold dev fails, print an error message suggesting a compatible base
+        image (e.g., `gcr.io/distroless/cc:debug`)
+
+    -   Default to `false`, and print a message suggesting to set it to
+        `true` when they run `skaffold dev`.
+
+2.  Should the URL used to download `watchexec` be exposed as a field in the
+    Skaffold schema?
+
+    This can be helpful for two reasons:
+
+    -   Users can specify a different version of `watchexec`, or the
+        musl-based binary instead of the default glibc one.
+
+    -   Some users work in environments with restricted internet access,
+        and this would allow them to specify an internal HTTP server as an
+        alternative.
+
+    However, an alternative that we can document is to ask users to download
+    `watchexec` out-of-band, and place it in the `kodata` directory before
+    running `skaffold dev`. Skaffold checks for the presence of the
+    `watchexec` binary in this directory and skips the download if it is
+    present. This option avoids adding yet another field to the schema.
+
+3.  Should the feature also cover the Docker deployer?
+
+## Alternative implementation steps
```

Comment:
I think the above suggestion vs this alternative is desirable (+1 to recommended implementation)